### DISCUSSION TITLE
Deploy Nextjs with Node js as backend

### DISCUSSION THREAD

**Discussion by malik-zee-code** [vercel/community#3419] [Help]
### Summary

Hey Guys,
When i deploy it to Vercel its back-end stops to work i know whats wrong with it that it doesn't take my node js back-end into consideration while making a build because of server.js file that was also not considered by next while making a build folder.

### Steps to Reproduce

i have been looking on the internet like how to deploy the nextjs with back-end as Nodejs in a single deployment like it won't be deployed each one of them. On a single NextJs deployment i wanted to deploy its back-end also i created a _server.js_ file to host my back-end and as well as front-end. it works fine on localhost. But when i deploy it to Vercel its back-end stops to work i know whats wrong with it that it doesn't take my node js back-end into consideration while making a build because of server.js file that was also not considered by next while making a build folder. This is my server js file
![image](https://github.com/vercel/community/assets/77394973/944a1684-fe60-426d-b4fa-e19118327f86)

How can i overcome this problem is there any way i can get it done or its not a good approach? what do you think ?

**Comment by denizsarikas** [comment]
Hello i am facing the exact same issue, i am useing next 14 and using my own server.js file in /app/server/server.js . but i could not find a way to make it work on the vercel. Did you find any solution for this problem?
